icx-proxy/src/main.rs

```diff
@@ -187,10 +239,51 @@ async fn forward_request(
             }
         }
 
-        Ok(builder.body(body)?)
+        (None, builder.body(body)?)
     } else {
-        Ok(builder.body(http_response.body.into())?)
+        (
+            if logger.is_trace_enabled() {
+                Some(http_response.body.clone())
+            } else {
+                None
+            },
+            builder.body(http_response.body.into())?,
+        )
+    };
+
+    if logger.is_trace_enabled() {
+        slog::trace!(
+            logger,
+            ">> {:?} {} {}",
+            &response.version(),
+            response.status().as_u16(),
+            response.status().to_string()
+        );
+
+        for (name, value) in response.headers() {
+            let value = String::from_utf8_lossy(value.as_bytes());
+            slog::trace!(logger, ">> {}: {}", name, value);
+        }
+
+        let body = body.unwrap_or_else(|| b"... streaming ...".to_vec());
+
+        slog::trace!(logger, ">>");
+        slog::trace!(
+            logger,
+            ">> {}{}",
+            match std::str::from_utf8(&body) {
+                Ok(s) => format!(r#""{}""#, s[..usize::min(100, s.len())].escape_default()),
+                Err(_) => hex::encode(&body[..usize::min(100, body.len())]),
+            },
+            if body.len() > 500 {
```

Comment:
Could we use the same cutoff here as we are using for the slice size above?
```suggestion
            if body.len() > 100 {
```

Reply:
Yeah and I should make it a constant.